### DISCUSSION TITLE
When using swr infinite how to clear the cache for all the pages?

### DISCUSSION THREAD

**Discussion by pedromcunha** [vercel/swr#2145] [Q&A]
Hi! Loving this project, really great work. I'm having some trouble clearing the data fetched by an swr infinite hook. I've been looking at setSize(0) but this just changes the page to 0, it doesn't actually update the cache to remove existing entries.

**Comment by huozhi** [comment]
If you’re using swr v2 beta, you can probably give this approach a try. You can use the matcher to match certain keys and reset the cache

https://github.com/vercel/swr/issues/2117#issuecomment-1213620333

**Reply by pedromcunha** [reply]
Thanks I'll give it a go!
